object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java

```diff
@@ -61,6 +92,73 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
     return super.visitMethodInvocation(node, p);
   }
 
+  @Override
+  public Void visitNewClass(NewClassTree node, Void p) {
+    if (!isAssignedToLocal(this.getCurrentPath())) {
+      TypeMirror type = TreeUtils.typeOf(node);
+      if (hasAlwaysCall(type)) {
+        checker.report(node, new DiagMessage(Diagnostic.Kind.ERROR, "missing.alwayscall", " "));
+      }
+    }
+    return super.visitNewClass(node, p);
+  }
+
+  private List<String> getCalledMethodAnnotation(MethodInvocationTree node) {
```

Comment:
Rather than this method, use the type factory's `getAnnotatedType` method and then `isSubtype`

Reply:
Should I make a dummy CalledMethod type out of AlwaysCall to use isSubtype?
Could you please clarify why you think it's the cleaner way? 
@kelloggm

Reply:
> Should I make a dummy CalledMethod type out of AlwaysCall to use isSubtype

Yes, that's what I had in mind.

> Could you please clarify why you think it's the cleaner way?

Sure. There are two reasons:
1. it avoids duplicating code, following the DRY ("don't repeat yourself) principle
2. it will enable these checks to take advantage of improvements in the `isSubtype` code. For example, if the type is an `@CalledMethodsPredicate` annotation and the checker was using an SMT solver to allow precise subtyping between those annotations and `@CalledMethods` annotations, then using `isSubtype` would allow the check here to succeed. If you re-implement, as you've done here, you wouldn't be able to take advantage of such improvements.